modules/porous_flow/src/materials/PorousFlowCapillaryPressureLinear.C

```diff
@@ -0,0 +1,37 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowCapillaryPressureLinear.h"
+
+template<>
+InputParameters validParams<PorousFlowCapillaryPressureLinear>()
+{
+  InputParameters params = validParams<PorousFlowCapillaryPressureBase>();
+  params.addRequiredParam<Real>("pc_max",  "Maximum capillary pressure (Pa)");
+  params.addClassDescription("This Material provides a linear capillary pressure");
+  return params;
+}
+
+PorousFlowCapillaryPressureLinear::PorousFlowCapillaryPressureLinear(const InputParameters & parameters) :
+  PorousFlowCapillaryPressureBase(parameters),
+  _pc_max(getParam<Real>("pc_max"))
+{
+}
+
+void
+PorousFlowCapillaryPressureLinear::computeQpProperties()
+{
+  /// Capillary pressure and derivatives wrt phase saturation at the nodes
```

Comment:
We already loop over all elements before the simulation to size certain
data structures. Look at ComputeMaxQps. We could probably store addition
pieces of data to make these types of checks more efficient during the
simulation.
On Mon, May 9, 2016 at 5:09 PM Daniel Schwen <notifications@github.com>
wrote:

> In modules/porous_flow/src/materials/PorousFlowCapillaryPressureLinear.C
> <https://github.com/idaholab/moose/pull/6889#discussion_r62588645>:
>
> > +  InputParameters params = validParams<PorousFlowCapillaryPressureBase>();
> > +  params.addRequiredParam<Real>("pc_max",  "Maximum capillary pressure (Pa)");
> > +  params.addClassDescription("This Material provides a linear capillary pressure");
> > +  return params;
> > +}
> > +
> > +PorousFlowCapillaryPressureLinear::PorousFlowCapillaryPressureLinear(const InputParameters & parameters) :
> > +  PorousFlowCapillaryPressureBase(parameters),
> > +  _pc_max(getParam<Real>("pc_max"))
> > +{
> > +}
> > +
> > +void
> > +PorousFlowCapillaryPressureLinear::computeQpProperties()
> > +{
> > +  /// Capillary pressure and derivatives wrt phase saturation at the nodes
>
> The minimum in my opinion should be an assert on the sizes right here. The
> actual check is not straightforward, as you could have mixed element types
> (varying number of nodes). So you cannot do a simple check in a
> constructor. I think you may have to iterate over the mesh. Comments
> @permcody <https://github.com/permcody>
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6889/files/4f2d0ae39e837aeec61169a353951bd97d029281#r62588645>
>